### ISSUE TITLE
About WIFI & Bluetooth upgrade

### ISSUE THREAD

**Issue by zhengqing-cn** [ameyrupji/thunderbolt-macpro-5-1#4]
Hi Ameyrupji,

First, I really appreciate your very detailed upgrade guidance, which really helps me a lot. However, I have a quick question about the WIFI & Bluetooth upgrade section. Is that possible to upgrade the Wifi and Bluetooth module without the bare cable? Since It is too ugly. Again, thank you so much!

Zheng

**Comment by ameyrupji** [comment]
I guess it can! You could keep your existing Bluetooth card but I think you will lose the ability to Airdrop! After a lot of reading and looking through numerous upgrade videos this is the cheapest and working upgrade! Don't worry about it being ugly you will not be able to see it once you put in your CPU tray and close off your case! I was also feeling the same but the convenience after having the card in is more important than the looks hands down!

**Comment by zhengqing-cn** [comment]
Yeah maybe it is meaningless to be bothered by this problem! Thank you for your response and again thank you for your detailed test especially on the thunderbolt 3 modules.

**Comment by zhengqing-cn** [comment]
Also, can I use any of the thunderbolt port to power back to the mac pro? I have a thunderbolt 3 expansion dock. It can charge my MacBook pro with about 90 watt. If I plug this device to the mac pro with thunderbolt 3 port, what gonna happened😂？

**Comment by ameyrupji** [comment]
Good question! Please connect with the fb group on this! The link to the fb group should be in the Useful links section in the bottom of the article! Someone must have tied doing this! I have not tried it myself! You are also welcome to contribute to this just create a PR from your fork and I will approve it so that others can also benefit from your findings on using an expansion dock!

**Comment by zhengqing-cn** [comment]
I just found this article and it seems like answered this question at some degree: https://lowendmac.com/2020/the-mac-pro-31-to-51-gets-thunderbolt-3/

**Comment by ameyrupji** [comment]
Perfect! looks like you have your answer!
